lib/CodeTransform/Adapter/WorseReflection/Transformer/CompleteConstructor.php

```diff
@@ -130,14 +130,41 @@ private function transformPromote(SourceCode $source): TextEdits
 
         foreach ($this->candidateClasses($source) as $class) {
             $constructMethod = $class->methods()->get('__construct');
+            $nonPromotedParameterNames = $this->getParentClassParamaterNames($class);
             foreach ($constructMethod->parameters()->notPromoted() as $parameter) {
+                if (in_array($parameter->name(), $nonPromotedParameterNames)) {
+                    continue;
+                }
                 $edits[] = TextEdit::create($parameter->position()->start()->toInt(), 0, sprintf('%s ', $this->visibility));
             }
         }
 
         return TextEdits::fromTextEdits($edits);
     }
 
+    /**
+    * Get the names of the parent constructor to know which parameters should not be promoted.
+    * @return array<string>
+    */
+    private function getParentClassParamaterNames(ReflectionClass $class): array
+    {
+        $ancestor = $class->ancestors()->firstOrNull();
```

Comment:
`parent()`? ancestors includes interfaces etc.

Reply:
Well we probably need a loop in here to see if the classes even higher up have constructors as well like so:
```php
class B {
    public function __construct(private string $a) {}
}
class A extends B {}

class Foo extends A {
    public function __construct(string $a) {parent::__construct($a);}
}
```

Then it wouldn't pick up on the constructor right?

Reply:
If you call `parent()->methods()->get('__construct')` it should return the first constructor in the heirarchy starting from the parent

Reply:
Added a test for it and it seems to pass.